### PR TITLE
feature/app-5493-score-chart

### PR DESCRIPTION
**Pull request by acassiovilasboas**
## Description

- add new component ScoreChart

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

Simulator Screenshot - iPhone 14 Pro - 2024-03-05 at 17 37 24

Simulator Screenshot - iPhone 14 Pro - 2024-03-05 at 17 37 28

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+
+extension OceanSwiftUI {
+    // MARK: Parameters
+
+    public class ScoreChartParameters: ObservableObject {
+        @Published public var title: String
+        @Published public var subtitle: String
+        @Published public var scoreMin: Double
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:17]: fica redundante colocar o score no nome da variavel
sugestao: minValue, maxValue e currentValue

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        private func configureChart() {
+            calculateDiff()
+            setPrimaryColorIfNeed()
+        }
+
+        private func calculateDiff() {
+            self.scoreDifference = self.scoreMax - self.scoreCurrent
+        }
+
+        private func setPrimaryColorIfNeed() {
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:53]: essa função pode ser dentro do scoreCharts retornando a cor, fica estranho preencher uma variavel que será utilizada na view

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            didSet {
+                self.configureChart()
+            }
+        }
+        @Published public var scoreCurrent: Double {
+            didSet {
+                self.configureChart()
+            }
+        }
+        @Published fileprivate var colorPrimary: UIColor = Ocean.color.colorStatusNegativePure
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:28]: No inglês o correto é o inverso, primaryColor e secondaryColor.

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        @Published fileprivate var colorSecondary: UIColor = Ocean.color.colorInterfaceLightDeep
+        @Published fileprivate var scoreDifference: Double = 0
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:30]: Acho que aqui se aplica o mesmo que o Vini mencionou de não colocar score novamente no nome.

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        private func configureChart() {
+            calculateDiff()
+            setPrimaryColorIfNeed()
+        }
+
+        private func calculateDiff() {
+            self.scoreDifference = self.scoreMax - self.scoreCurrent
+        }
+
+        private func setPrimaryColorIfNeed() {
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:53]: O texto correto é ifNeeded, faltou o "ed" no final

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift (hunk 1 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:17]: hahahahahahha, estava exatamente esses nomes

File: OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift (hunk 5 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/ScoreChart/OceanSwiftUI+ScoreChart.swift:53]: Esse nome nao faz mais sentido... retirado.

**Review by renanmassaroto — APPROVED** [review]

**Review by vromeiro — APPROVED** [review]

**Pull request merged**